teamvault/apps/secrets/templates/secrets/detail_content/password.html

```diff
@@ -102,8 +102,19 @@ <h5 class="modal-title" id="changeSecretModalLabel">{% translate "Secret needs c
         })
 
         function largeType(password) {
+            const colorfulPasswordHTML = password.split("").map(character => {
+                // digits
+                if(/\d/.test(character)) return `<span class="text-danger">${character}</span>`;
```

Comment:
Wenn schon regex dann vielleicht lieber .replaceAll um das Ersetzen direkt in der Regex-Engine zu machen:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll

Alternativ kann man das auch mit einfachen Checks basierend auf dem ASCII Code machen:
```
if(character >= '0' && character <= '9') { number }
if(character >= 'a' && character <= 'z' || ) { small char }
…
```

Ich fände es gut auf dem Weg auch kleine und große Buchstaben zu unterscheiden (kleines L vs. großes i)

Reply:
Mit `replaceAll` funktioniert der erste Zeichentyp gut, allerdings kommen wir beim zweiten Durchlauf mit den existierenden span Tags in die quere.

Reply:
🚨 Außerdem habe ich gerade germerkt, dass ich durch `innerHTML` eine Code Injection erzeugt habe. Wir sollten entweder alle Zeichen wie `<`, `>` escapen oder die Elemente programmatisch mit `createElement` hinzufügen.